ui/elastic/src/App.js

```diff
@@ -89,9 +89,33 @@ const HitsTable = ({ hits }) => {
 
 const sortVersions = (filters) => {
   return filters.sort((a, b) => {
-    const majorA = parseInt(a.key.split(".", 1)[0], 10)
-    const majorB = parseInt(b.key.split(".", 1)[0], 10)
-    return majorB - majorA
+    const partsA = a.key.split(".")
+    const partsB = b.key.split(".")
+    if (partsA[0] !== partsB[0]) {
+      // sort on major version.
+      return parseInt(partsB[0], 10) - parseInt(partsA[0], 10)
+    }
+    if (partsA[1] !== partsB[1]) {
+      const subPartRegex = /^(\d+)([a-zA-Z]+)?(\d+)?([a-zA-Z]+)?/ // Eg: 0b12pre
```

Comment:
Yeah, i'm trying to balance between "implement all the things so it's perfect" and "make it as simple as possible but still work for very old/legacy versions". For this current case, it seems it just "works well":

<img width="252" alt="screen shot 2017-08-01 at 09 47 18" src="https://user-images.githubusercontent.com/167767/28814581-78e84e16-769e-11e7-9e2c-5fe29a6174f2.png">

So I'm not sure I should complicate things a bit more? Another solution would be to use the same "clever parsing for the part after the first period" for each part, and recurse/loop over all the parts. Would that result in cleaner and more understandable code?